### PR TITLE
[HWKMETRICS-545] [HWKMETRICS-502] Compression job control

### PR DESCRIPTION
**Pull request by burmanm**
This PR adds the ability of manually adjusting the blockSize of the compression (accepting the ISO8601 format as parameter) as well as creating MBean for the compression job triggering for given timestamp.

Calculation of the compressed range is moved from MetricsServiceImpl to CompressData to allow customized ranges to be triggered.

### CODE REVIEW

**Comment by burmanm** [comment]
@jsanda I'm seeing some issue with scheduler occasionally (does not happen every time), where it does not actually schedule the manual run at all.

https://gist.github.com/burmanm/56af20a0ae2b22a98791e291c8481b62

**Comment by jsanda** [comment]
Can you change ```submitCompressJob(Map<String, String> parameters)``` to subscribe to the returned Single and log the results? If an error occurs scheduling the job, it is just getting swallowed.

**Comment by burmanm** [comment]
@jsanda Yes, I can try debugging that way. Should I however modify the original scheduleJob to give it .doOnError() instead of making each caller do the subscribe + logging just to get stats?

**Comment by jsanda** [comment]
> Should I however modify the original scheduleJob to give it .doOnError() instead of making each caller do the subscribe + logging just to get stats?

That sounds good. We could use some INFO or DEBUG statements in SchedulerImpl.scheduleJob() that logs the outcome.

**Comment by FilipB** [comment]
retest this please

File: core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsService.java [hunk 1]
```diff
@@ -171,7 +171,17 @@
     <T> Observable<DataPoint<T>> findDataPoints(MetricId<T> id, long start, long end, int limit, Order order,
             int pageSize);
 
-    Observable<Void> compressBlock(Observable<? extends MetricId<?>> metrics, long timeSlice, int pageSize);
```
Comment (jsanda) [inline core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsService.java]: Why not return Completable here? One of its primary use cases was to address Observable<Void>. I know that we have plenty of places where we use Observable<Void>, but I would like to eventually replace them with Completable provided we only care about a terminal event which is almost always the case.

**Comment by jsanda** [comment]
Could you add a test that demonstrates scheduling a compression job via JMX?

**Comment by burmanm** [comment]
That would require creating some sort of access methods to the job scheduler. As the JMX is initialized in the Wildfly phase, it requires the whole Wildfly to be running and only then can I start doing the JMX requests. So same sort of packaging as the rest-tests and then request the scheduler's statistics I guess..

Or Arquillian perhaps. Ugh.

**Comment by burmanm** [comment]
With blocking processing I haven't seen missed schedulings.

**Pull request merged**